services/telemetry/src/util.js

```diff
@@ -1,23 +1,55 @@
-/**
- * Sends a Protobuf message or its JSON representation,
- * based on whether or not the request's Accept header
- * is `application/json`.
- */
-export function sendJsonOrProto(req, res, proto) {
-    const accept = req.get('accept');
-
-    if (accept === undefined || !accept.startsWith('application/json')) {
-        res.set('content-type', 'application/x-protobuf');
-        res.send(Buffer.from(proto.serializeBinary()));
-    } else {
-        res.send(proto.toObject());
-    }
+/** Convert radians to degrees. */
+export function degrees(radians) {
+  return radians * 180 / Math.PI;
 }
 
-/**
- * Returns a Python-like remainder.
- * https://stackoverflow.com/a/3417242
- */
-export function wrapIndex(i, i_max) {
-    return ((i % i_max) + i_max) % i_max;
-}
\ No newline at end of file
+/** Put degrees in range [0, 360). */
+export function modDegrees(degrees) {
+  return (degrees % 360 + 360) % 360;
+}
+
+/** Put degrees in range (-180, 180]. */
+export function modDegrees2(degrees) {
+  return -modDegrees(-degrees - 180) + 180;
+}
+
+/** Return a object that can be used for cleanup in timeouts. */
+export function cleanupAfter(cleanup, limit, message) {
+  let timeoutMessage = message || 'operation timed out';
+
+  let earlyCall = false;
+  let earlyErr = null;
+
+  // Store if there was an error or not in the case that the finish
+  // function is called before it is overrided below.
+  let finish = (err) => {
+    earlyCall = true;
+    earlyErr = err;
+  };
+
+  // Resolves after the cleanup is finished.
+  let done = new Promise((resolve, reject) => {
+    // Call the finish function if the timeout is reached.
+    let waitTimeout = setTimeout(() => {
+      finish(Error(timeoutMessage));
+    }, limit);
+
+    // Cancel the timeout, cleanup, and mark the operation as
+    // complete.
+    finish = (err, result) => {
+      clearTimeout(waitTimeout);
+      cleanup();
+
+      if (err) reject(err);
+      else resolve(result);
+    };
+
+    if (earlyCall) finish(earlyErr);
+  });
+
+  // Async function to wait for the operation to be over and cleaned
+  // up.
+  let wait = async () => await done;
```

Comment:
Since all the other non-private functions which return Promises are marked with async/await, and because if you were to pass this wait function into the task queue with [async.js](http://caolan.github.io/async:) it is required that it a [native async function](http://caolan.github.io/async/index.js.html#line27) for it to work, as opposed to just a function which returns a promise. The async/await vs a function that returns a Promise distinction can be a problem in the future with dealing with the task queue if one doesn't know about this.